lib/modulesync.rb

```diff
@@ -177,4 +178,83 @@ def self.update(cli_options)
     end
     exit 1 if errors && options[:fail_on_warnings]
   end
+
+  def self.clone(cli_options)
+    @options = config_defaults.merge(cli_options)
+
+    managed_modules.each do |puppet_module|
+      puppet_module.repository.clone unless puppet_module.repository.cloned?
+    end
+  end
+
+  def self.execute(cli_options)
+    @options = config_defaults.merge(cli_options)
+
+    errors = {}
+    managed_modules.each do |puppet_module|
+      $stdout.puts "#{puppet_module.given_name}:"
+
+      puppet_module.repository.clone unless puppet_module.repository.cloned?
+      puppet_module.repository.switch branch: @options[:branch]
+
+      command_args = cli_options[:command_args]
+      local_script = File.expand_path command_args[0]
+      command_args[0] = local_script if File.exist?(local_script)
+
+      # Remove bundler-related env vars to allow the subprocess to run `bundle`
+      command_env = ENV.reject { |k, _v| k.match?(/(^BUNDLE|^SOURCE_DATE_EPOCH$|^GEM_|RUBY)/) }
+
+      FileUtils.chdir(puppet_module.working_directory) do
+        result = system command_env, *command_args, unsetenv_others: true
```

Comment:
Rather than using `chdir`, you can pass `chdir` to system
```suggestion
      result = system(command_env, *command_args, unsetenv_others: true, chdir: puppet_module.working_directory)
```